test/parallel/rolldown.withdraw.merkleRoot.test.ts

```diff
@@ -0,0 +1,472 @@
+  await Sudo.batchAsSudoFinalized(
+    Assets.mintToken(ETH_ASSET_ID, testUser, BN_MILLION),
+  );
+  await testUser.refreshAmounts(AssetWallet.BEFORE);
+  await signTx(
+    getApi(),
+    await Withdraw(testUser, BN_HUNDRED, ethIdL1Asset.ethereum, "Ethereum"),
+    testUser.keyRingPair,
+  ).then((events) => {
+    const res = getEventResultFromMangataTx(events);
+    expect(res.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+  });
+  await Rolldown.waitForNextBatchCreated("Ethereum", waitingBatchPeriod);
+  await testUser.refreshAmounts(AssetWallet.AFTER);
+  const withdrawalAmount = testUser
+    .getAsset(ETH_ASSET_ID)!
+    .amountBefore.free.sub(testUser.getAsset(ETH_ASSET_ID)!.amountAfter.free);
+  expect(withdrawalAmount).bnEqual(BN_HUNDRED);
+});
+
+test("GIVEN <batchSize - 1> withdraws, AND manually generated batch and create another withdraw for some other network (arb) THEN the batch is not generated", async () => {
+  let error: any;
+  //since there is no token in the Arbitrum chain by default, we create a new one
+  const minToBeSequencer = await SequencerStaking.minimalStakeAmount();
+  const blocksForSequencerUpdate =
+    await SequencerStaking.blocksForSequencerUpdate();
+  await clearActiveSequencers();
+  await signTx(
+    await getApi(),
+    await SequencerStaking.provideSequencerStaking(
+      minToBeSequencer.addn(1234),
+      "Arbitrum",
+    ),
+    testUser.keyRingPair,
+  );
+  await Rolldown.waitForReadRights(
+    testUser.keyRingPair.address,
+    50,
+    "Arbitrum",
+  );
+  const txIndex = await Rolldown.lastProcessedRequestOnL2("Arbitrum");
+  const api = getApi();
+  const update = new L2Update(api)
+    .withDeposit(
+      txIndex,
+      testUser.keyRingPair.address,
+      testUser.keyRingPair.address,
+      BN_MILLION,
+    )
+    .on("Arbitrum")
+    .buildUnsafe();
+  await signTx(api, update, testUser.keyRingPair).then((events) => {
+    const res = getEventResultFromMangataTx(events);
+    expect(res.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+  });
+  await waitForNBlocks(blocksForSequencerUpdate);
+
+  //we need to run <batchSize> extrinsic to update the start of the automatic batching period
+  await Sudo.batchAsSudoFinalized(
+    ...(await getWithdrawalExtrinsics(batchSize)),
+  );
+
+  await Sudo.batchAsSudoFinalized(
+    ...(await getWithdrawalExtrinsics(batchSize - 1)),
+  );
+
+  await signTx(
+    getApi(),
+    await Withdraw(
+      testUser,
+      BN_HUNDRED,
+      testUser.keyRingPair.address,
+      "Arbitrum",
+    ),
+    testUser.keyRingPair,
+  ).then((events) => {
+    const res = getEventResultFromMangataTx(events);
+    expect(res.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+  });
+  try {
+    await Rolldown.waitForNextBatchCreated("Ethereum", 10);
+  } catch (err) {
+    error = err;
+  }
+  expect(error).toEqual(
+    "method rolldown.TxBatchCreated not found within blocks limit",
+  );
+});
+
+describe("Pre-operation withdrawal tests", () => {
```

Comment:
if a test file has a describe, please create another describe for the other tests above ^^

Reply:
Added